Using non semver versions break in 2.18.5

### I tried this:

When using non-semver versioning (just 1,2,3) catalog creation breaks since 2.18.5 -> https://github.com/event-catalog/eventcatalog/commit/eb019818ef1c0c8582f571c130b50cc93cc3dd77

### This happened:

```
12:17:54 [ERROR] [build] Failed to call getStaticPaths for src/pages/discover/[type]/index.astro
Invalid Version: 1
  Location:
    /builds/..../eventcatalog/node_modules/semver/classes/semver.js:38:13
  Stack trace:
    at new SemVer (/builds/..../eventcatalog/node_modules/semver/classes/semver.js:38:13)
    at file:///builds/..../eventcatalog/.eventcatalog-core/.astro/chunks/Seo_B5oJq-mu.mjs:551:12
    at getVersionForCollectionItem (file:///builds/..../eventcatalog/.eventcatalog-core/.astro/chunks/Seo_B5oJq-mu.mjs:550:36)
```

### I expected this:

Would be nice if non-semver versions are still supported :)

### Is there a workaround?

_No response_

### Anything else?

_No response_

### EventCatalog Version

2.18.5 (6)

### Node.js Version

22

### Platform(s)

Linux

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

--- a/.changeset/four-socks-tap.md
+++ b/.changeset/four-socks-tap.md
@@ -1,0 +1,5 @@
+---
+"@eventcatalog/core": patch
+---
+
+fix(core): added ability to sort by semver and numbers

--- a/eventcatalog/src/utils/__tests__/collections/util.spec.ts
+++ b/eventcatalog/src/utils/__tests__/collections/util.spec.ts
@@ -27,11 +27,12 @@ describe('Collections - utils', () => {
   });
 
   describe('findLatestVersion', () => {
-    it.each([[{ versions: ['1', '3', '2'], latest: '3' }], [{ versions: ['1.0.1', '1.1.0', '1.0.2'], latest: '1.1.0' }]])(
-      'should returns $latest as latest version of $versions',
-      ({ versions, latest }) => {
-        expect(findLatestVersion(versions)).toBe(latest);
-      }
-    );
+    it.each([
+      [{ versions: ['1', '3', '2'], latest: '3' }],
+      [{ versions: ['1.0.1', '1.1.0', '1.0.2'], latest: '1.1.0' }],
+      [{ versions: ['a', 'c', 'b'], latest: 'c' }],
+    ])('should returns $latest as latest version of $versions', ({ versions, latest }) => {
+      expect(findLatestVersion(versions)).toBe(latest);
+    });
   });
 });

--- a/eventcatalog/src/utils/__tests__/collections/util.spec.ts
+++ b/eventcatalog/src/utils/__tests__/collections/util.spec.ts
@@ -1,4 +1,4 @@
-import { satisfies } from '@utils/collections/util';
+import { findLatestVersion, satisfies } from '@utils/collections/util';
 import { describe, it, expect } from 'vitest';
 
 describe('Collections - utils', () => {
@@ -24,5 +24,14 @@ describe('Collections - utils', () => {
     ])('should returns $expected to version as $version and range as $range', ({ expected, version, range }) => {
       expect(satisfies(version, range)).toBe(expected);
     });
+  });
+
+  describe('findLatestVersion', () => {
+    it.each([[{ versions: ['1', '3', '2'], latest: '3' }], [{ versions: ['1.0.1', '1.1.0', '1.0.2'], latest: '1.1.0' }]])(
+      'should returns $latest as latest version of $versions',
+      ({ versions, latest }) => {
+        expect(findLatestVersion(versions)).toBe(latest);
+      }
+    );
   });
 });

--- a/eventcatalog/src/utils/collections/util.ts
+++ b/eventcatalog/src/utils/collections/util.ts
@@ -14,24 +14,29 @@ export const getVersions = (data: CollectionEntry<CollectionTypes>[]) => {
   return { versions, latestVersion };
 };
 
+export function findLatestVersion(versions: string[]) {
+  // try to coerce semver versions from string input
+  const semverVersions = versions.map((v) => ({ original: v, semver: coerce(v) }));
+
+  // if all versions are semver'ish, use semver to order the versions
+  if (semverVersions.every((v) => v.semver != null)) {
+    return semverVersions.sort((a, b) => compare(b.semver!, a.semver!))[0].original;
+  } else {
+    // fallback to default sort
+    return versions.sort()[0];
+  }
+}
+
 export const getVersionForCollectionItem = (
   item: CollectionEntry<CollectionTypes>,
   collection: CollectionEntry<CollectionTypes>[]
 ) => {
-  const allVersionsForItem = collection
-    .filter((i) => i.data.id === item.data.id)
-    .map((i) => i.data.version)
-    .sort();
+  const allVersionsForItem = collection.filter((i) => i.data.id === item.data.id).map((i) => i.data.version);
 
   // unique versions
   const versions = [...new Set(allVersionsForItem)];
 
-  // Use semver to order the versions
-  const orderedVersions = versions.sort((a, b) => {
-    return compare(b, a);
-  });
-
-  const latestVersion = orderedVersions[0];
+  let latestVersion = findLatestVersion(versions);
 
   return { versions, latestVersion };
 };

--- a/eventcatalog/src/utils/collections/util.ts
+++ b/eventcatalog/src/utils/collections/util.ts
@@ -23,7 +23,7 @@ export function findLatestVersion(versions: string[]) {
     return semverVersions.sort((a, b) => compare(b.semver!, a.semver!))[0].original;
   } else {
     // fallback to default sort
-    return versions.sort()[0];
+    return versions.sort().reverse()[0];
   }
 }
 

